mono/mini/interp/transform.c

```diff
@@ -2927,21 +2927,28 @@ generate (MonoMethod *method, MonoMethodHeader *header, InterpMethod *rtm, unsig
 				if (m_class_get_parent (klass) == mono_defaults.array_class) {
 					ADD_CODE(td, MINT_NEWOBJ_ARRAY);
 					ADD_CODE(td, get_data_item_index (td, mono_interp_get_imethod (domain, m, error)));
-				} else if (!m_class_is_valuetype (klass) &&
-						klass != mono_defaults.string_class &&
+				} else if (klass != mono_defaults.string_class &&
 						!mono_class_is_marshalbyref (klass) &&
 						!mono_class_has_finalizer (klass) &&
 						!m_class_has_weak_fields (klass)) {
-					MonoVTable *vtable;
 					MonoMethodSignature *csig;
-					vtable = mono_class_vtable_checked (domain, klass, error);
-					goto_if_nok (error, exit);
-					csig = mono_method_signature (m);
 
-					ADD_CODE(td, MINT_NEWOBJ_FAST);
+					if (!m_class_is_valuetype (klass))
+						ADD_CODE(td, MINT_NEWOBJ_FAST);
+					else if (mint_type (m_class_get_byval_arg (klass)) == MINT_TYPE_VT)
+						ADD_CODE(td, MINT_NEWOBJ_VTST_FAST);
```

Comment:
`VTST`? does it mean `VTSTACK`?

Reply:
@lewurm yes. Do you prefer VTSTACK in name ?